### DISCUSSION TITLE
Make static implicit on all members in in C# static classes

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/csharplang#3967] [Language Ideas]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/1201344/make-static-implicit-on-all-members-in-in-c-static.html)._

---
<p>When I create a static class, the static should be implicitly applied, not merely required, on all members.  Being forced to put static on every member in a static class is silly redundancy for a language that strives for clarity via brevity.  I should not have to write</p><pre>static class Utility {
public static int SomeFunc() {...}
}</pre><p>I should be able to just write</p><pre>static class Utility {
public int SomeFunc() {...}
}</pre>

---
#### Mila Zhou[MSFT] on 9/27/2020, 00:32 AM: 

Thank you for taking the time to provide your suggestion.  We will do some preliminary checks to make sure we can proceed further.  We'll provide an update once the issue has been triaged by the product team.

**Comment by Joe4evr** [comment]
The reason `static` is required to be specified on all members of a static class is because removing `static` from the class declaration in a future version is a binary-compatible change. If `static` was implicit on its members, you'd have to remember to add it to everything when you remove it from the class to avoid breaking your consumers.
